### PR TITLE
feat(theme-common): code block MagicComments support for TeX/LaTeX/Matlab

### PR DESCRIPTION
**Pull request by jj-style**
## Pre-flight checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [x] **If this is a code change**: I have written unit tests and/or added dogfooding pages to fully verify the new behavior.
- [ ] **If this is a new API or substantial change**: the PR has an accompanying issue (closes #0000) and the maintainers have approved on my working plan.

## Motivation

I could not see an open issue for it but I would like to be able to use the magic comments feature in Matlab code blocks. This isn't currently supported due to Matlab's way of commenting, therefore I have added this capability in this PR.

## Test Plan

- I have added a Matlab code snippet in the
 dogfooding page which verifies the functionality.
- I ran a local instance of the site and verified that the lines were not highlighted by the magic comments
- I did the implementation and restarted the server and verified that the lines were highlighted

before
before
after
after

### Test links

Deploy preview: https://deploy-preview-8982--docusaurus-2.netlify.app/tests/pages/code-block-tests/#magic-comments-tests

### CODE REVIEW

**Review by Josh-Cena — COMMENTED** [review]
Thanks! This pattern is inherited from TeX so let's call the style `tex` and add `tex` and `latex` to the language list.

**Comment by jj-style** [comment]
Thank you for the feedback!
I have now changed the comment style to be called `tex` and added an example in dogfooding code blocks page to verify this for `latex` also

File: website/_dogfooding/_pages tests/code-block-tests.mdx [hunk 1]
```diff
@@ -322,3 +322,29 @@ WHERE customer_id IN (
...
+  result = n * 2;
+end
+% highlight-end
+
+x = 10;
+% highlight-next-line
+y = times2(x);
+```
+
+```latex title="latex.tex
```
Comment (slorber) [inline website/_dogfooding/_pages tests/code-block-tests.mdx:338]:
```suggestion
```latex title="latex.tex"
```

Missing close quote

**Comment by jj-style** [comment]
Thanks for the review and for pushing up the missing quote! 😊

**Pull request merged**